### DISCUSSION TITLE
Compiler without proper std::fs support

### DISCUSSION THREAD

**Discussion by ErikTempelaarVO** [open62541pp/open62541pp#263] [Q&A] [answered]
I have a question regarding the compile-time check to determine if std::fs is experimental / supported.

https://github.com/ErikTempelaarVO/open62541pp/blob/7e75bd39080832837cce87b09bc076500a783adb/include/open62541pp/types/Builtin.h#L12

I believe the logic is a bit flawed, because if the compiler doesn't have any support, it assumes full support.
Which is wrong for my compiler-variant, as already mentioned here:
https://github.com/open62541pp/open62541pp/pull/110#issuecomment-1807794052

````
#if !__has_include(<filesystem>) && __has_include(<experimental/filesystem>)
#include <experimental/filesystem>
namespace fs = std::experimental::filesystem;
#else
#include <filesystem>
namespace fs = std::filesystem;
#endif
````

I create a workaround for my compiler here:
https://github.com/open62541pp/open62541pp/commit/e5e8a6c4fefe1ec21751b6181ad35068d7af6cc3

Is it feasible to add an option to the library that disables any use of std::fs?

**Comment by lukasberbuer** [comment]
Hi @ErikTempelaarVO, what compiler are you using?
The library is using many C++17 features, so I can't imagine, that a compiler without C++17 support can compile this?
We can add a compile option, e.g. `UAPP_ENABLE_FILESYSTEM` to disable all fs related functions and imports.

**Reply by ErikTempelaarVO** [reply]
I use gcc 7.2.0 and it supports c++17, but on Windows (msys2/mingw32) it comes without any support for std::fs, the experimental support is not enabled by default. It requires an option 'enable-libstdcxx-filesystem-ts'.
(Non-experimental support for std::fs made it into gcc 8)

**Comment by ErikTempelaarVO** [comment, marked as answer]
Fixed by #266
